gamify-impl/src/main/java/ch/heigvd/gamify/ui/api/endpoints/BadgesController.java

```diff
@@ -0,0 +1,113 @@
+public class BadgesController implements BadgesApi {
+
+  @Autowired
+  ServletRequest request;
+
+  @Autowired
+  BadgeRepository badgeRepository;
+
+  @Autowired
+  CategoryRepository categoryRepository;
+
+  private final int LOWER_DEFAULT=0;
+  private final int UPPER_MAX =Integer.MAX_VALUE;
+
+  @Override
+  public ResponseEntity<List<Badge>> getBadges(@Valid Integer page, @Valid Integer size) {
+    List<Badge> bd=StreamSupport.stream(badgeRepository
+        .findAllByIdBadge_App((App) request.getAttribute(ApiKeyFilter.APP_KEY))
+        .spliterator(), false)
+        .map(BadgesController::toDto)
+        .collect(Collectors.toList());
+    if(bd.isEmpty()){
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+    return ResponseEntity.ok(bd);
+  }
+
+  @Override
+  public ResponseEntity<Badge> getBadge(String name) {
+    var app = (App) request.getAttribute(ApiKeyFilter.APP_KEY);
+    return badgeRepository.findById(BadgeIdentifier.builder()
+        .app(app).badgeName(name).build())
+        .map(BadgesController::toDto)
+        .map(ResponseEntity::ok)
+        .orElse(ResponseEntity.notFound().build());
+  }
+
+  @Transactional
+  @Override
+  public ResponseEntity<Void> putBadge(String name, @Valid Badge badge) {
+    if (!badge.getName().equals(name)) {
+      return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+    var app = (App) request.getAttribute(ApiKeyFilter.APP_KEY);
+    var category = categoryRepository.findById(CategoryIdentifier.builder()
+        .name(badge.getCategory())
+        .app(app).build());
+    if (category.isEmpty()) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+
+    badgeRepository.save(ch.heigvd.gamify.domain.badges.Badge.builder()
+        .idBadge(BadgeIdentifier.builder()
+            .app(app).badgeName(name).build())
+        .title(badge.getTitle())
+        .description(badge.getDescription())
+        .category(category.get())
+        .pointsLower(badge.getPointsLower().orElse(LOWER_DEFAULT))
+        .pointsUpper(badge.getPointsUpper().orElse(UPPER_MAX))
+        .build());
+    return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
+  }
+
+  @Transactional
+  @Override
+  public ResponseEntity<Void> deleteBadge(String name) {
+    var app = (App) request.getAttribute(ApiKeyFilter.APP_KEY);
+    var exists = badgeRepository.existsById(BadgeIdentifier
+    .builder().app(app).badgeName(name).build());
+    if (exists) {
+      badgeRepository.deleteById(BadgeIdentifier
+          .builder().app(app).badgeName(name).build());
+      return ResponseEntity.noContent().build();
+    } else {
+      return ResponseEntity.notFound().build();
+    }
+  }
+
+  private static Badge toDto(ch.heigvd.gamify.domain.badges.Badge badge) {
+    Badge badgeTemp = new Badge()
+        .title(badge.getTitle()==null ? "t": badge.getTitle())
+        .description(badge.getDescription()==null ? "d": badge.getDescription())
```

Comment:
Can these be null ? I'm not sure we want to provide default values.

Reply:
No, they can't be null, it's in the spec

Reply:
And can `badge.getTitle()` or `badge.getDescription()` be `null` then ?